common/lib/xmodule/xmodule/js/src/video/035_video_accessible_menu.js

```diff
@@ -1,241 +1,76 @@
-                    // open.
-                case KEY.DOWN:
-                    this.nextMenuItemLink(this.menuItemsLinks, index).focus();
-                    break;
-                    // Close menu.
-                case KEY.TAB:
-                    this.closeMenu();
-                        // TODO
-                        // What has to happen here? In speed menu, tabbing backward
-                        // will give focus to Play/Pause button and tabbing
-                        // forward to Volume button.
-                    break;
-                    // Close menu, give focus to button and change
-                    // file type.
-                case KEY.ENTER:
-                case KEY.SPACE:
-                    this.button.focus();
-                    this.changeFileType.call(this, event);
-                    this.closeMenu();
-                    break;
-                    // Close menu and give focus to speed control.
-                case KEY.ESCAPE:
-                    this.closeMenu();
-                    this.button.focus();
-                    break;
-                }
-                return false;
-            }
-            else {
-                switch (keyCode) {
-                    // Open menu and focus on last element of list above it.
-                case KEY.ENTER:
-                case KEY.SPACE:
-                case KEY.UP:
-                    this.openMenu();
-                    this.menuItemsLinks.last().focus();
-                    break;
-                    // Close menu.
-                case KEY.ESCAPE:
-                    this.closeMenu();
-                    break;
-                }
-                // We do not stop propagation and default behavior on a TAB
-                // keypress.
-                return event.keyCode === KEY.TAB;
-            }
-                                                },
-
-        setValue: function(value) {
-            this.value = value;
-            this.menuItems
-                .removeClass('active')
-                .find("a[data-value='" + value + "']")
-                .parent()
-                .addClass('active');
         },
 
         changeFileType: function(event) {
             var fileType = $(event.currentTarget).data('value'),
-                data = {'transcript_download_format': fileType};
-
-            this.setValue(fileType);
-            this.options.storage.setItem('transcript_download_format', fileType);
+                data = {transcript_download_format: fileType},
+                url = $(event.currentTarget).attr('href'),
+                that = this;
 
             $.ajax({
-                url: this.options.saveStateUrl,
+                async: false,
+                url: that.options.saveStateUrl,
                 type: 'POST',
                 dataType: 'json',
-                data: data
+                data: data,
+                success: function() {
```

Comment:
Previously, when a user used the dropdown menu to change the transcript type, a `POST` event was sent updating the local storage and storing the filetype selection. The same AJAX call is here, but I've set callbacks and made it synchronous because everything happens in a single click now, as opposed to a two-step process before (select filetype from dropdown, then click download button).